### PR TITLE
Lesson 1: Iterations

### PR DESCRIPTION
**Pull request by HcsOmot**
Tests are passing, code works and is correct.
But I need to redo this assignment, b/c the test clearly asked for iterations, I have delivered none (explicitly).

### CODE REVIEW

File: tests/BinaryGapTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     * @dataProvider binaryGap
+     */
+    public function testFindBinaryGapForInteger($integer, $binaryGap)
+    {
+        $binaryGapCalculator = new BinaryGapCalculator();
+
+        $this->assertEquals($binaryGap, $binaryGapCalculator->findBinaryGapForInteger($integer));
+    }
+
+    public function binaryGap(): array
```
Comment (msvrtan) [inline tests/BinaryGapTest.php]: Give some more meaningfull name to this data provider

File: tests/BinaryGapTest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        return [
+            [9, 2],
```
Comment (msvrtan) [inline tests/BinaryGapTest.php]: Please add comment how the number looks in binary form (for easier understanding)
